feature: add Custom Field doc and TS examples (#320)
vaadin/docs: articles/ds/components/custom-field/index.asciidoc

= Custom Field

:components-path-prefix: ../

// tag::description[]

> Custom Field is a component for wrapping multiple components as a single field. It has the same features as <<{components-path-prefix}input-fields#,Input Fields>>, such as its own label, helper, validation and data binding. Use it to create custom input components.

Reviewer: ```suggestion
Custom Field is a component for wrapping multiple components as a single field. It has the same features as <<{components-path-prefix}input-fields#,Input Fields>>, such as its own label, helper, validation, and data binding. Use it to create custom input components.
```

- https://vaadin.com/components/vaadin-custom-field/html-examples[Web Component Examples]
- https://vaadin.com/components/vaadin-custom-field/java-examples[Java Component Examples]
====

== Basic Usage

Custom Field is optimised for wrapping the following components:

* <<../text-field#,Text Field>>
* <<../number-field#,Number Filed>>
* <<../password-field#,Password Field>>
* <<../text-area#,Text Area>>
* <<../select#,Select>>
* <<../combo-box#,Combo Box>>
* <<../date-picker#,Date Picker>>
* <<../time-picker#,Time Picker>>

> It can also be used to provide a label, helper, etc. for components that don’t have a label by default, such as List Box.

Reviewer: ```suggestion
It can also be used to provide a label, helper, and the other field features, for components that don’t have them built-in, such as <<../list-box#,List Box>>.
```
